contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,334 @@
+    uint256 public immutable targetChainId;
+
+    ////////////////////////////////////////////////////
+    /// --- STORAGE
+    ////////////////////////////////////////////////////
+    /// @notice Fee in BASE_FEE unit payed when managing campaign.
+    uint16 public fee;
+
+    /// @notice Address of the fee collector
+    address public feeCollector;
+
+    /// @notice Id of the campaign created
+    uint256 public campaignId;
+
+    ////////////////////////////////////////////////////
+    /// --- EVENTS
+    ////////////////////////////////////////////////////
+    event FeeUpdated(uint16 _newFee);
+    event FeeCollected(address _feeCollector, uint256 _feeAmount);
+    event FeeCollectorUpdated(address _newFeeCollector);
+    event CampaignIdUpdated(uint256 _newId);
+    event BribeCreated(
+        address gauge,
+        address rewardToken,
+        uint256 maxRewardPerVote,
+        uint256 totalRewardAmount
+    );
+    event TotalRewardAmountUpdated(uint256 extraTotalRewardAmount);
+    event NumberOfPeriodsUpdated(uint8 extraNumberOfPeriods);
+    event RewardPerVoteUpdated(uint256 newMaxRewardPerVote);
+    event TokensRescued(address token, uint256 amount, address receiver);
+
+    ////////////////////////////////////////////////////
+    /// --- CONSTRUCTOR && INITIALIZATION
+    ////////////////////////////////////////////////////
+    constructor(
+        uint256 _targetChainId,
+        address _campaignRemoteManager,
+        address _rewardToken,
+        address _gauge
+    ) {
+        targetChainId = _targetChainId;
+        campaignRemoteManager = _campaignRemoteManager;
+        rewardToken = _rewardToken;
+        gauge = _gauge;
+
+        // Prevent implementation contract to be governed
+        _setGovernor(address(0));
+    }
+
+    /// @notice initialize function, to set up initial internal state
+    /// @param _strategist Address of the strategist
+    /// @param _fee Fee in BASE_FEE unit payed when managing campaign
+    /// @param _feeCollector Address of the fee collector
+    function initialize(
+        address _strategist,
+        uint16 _fee,
+        address _feeCollector
+    ) external onlyGovernor initializer {
+        _setStrategistAddr(_strategist);
+        _setFee(_fee);
+        _setFeeCollector(_feeCollector);
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- MUTATIVE FUNCTIONS
+    ////////////////////////////////////////////////////
+    /// @notice Create a new campaign on VotemarketV2
+    /// @dev This will use all token available in this contract
+    /// @param numberOfPeriods Duration of the campaign in weeks
+    /// @param maxRewardPerVote Maximum reward per vote to distribute, to avoid overspending
+    /// @param blacklist  List of addresses to exclude from the campaign
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function createCampaign(
```

Comment:
this one should have `nonReentrant` on it

Reply:
Done in this commit: https://github.com/OriginProtocol/origin-dollar/pull/2327/commits/8d33a72e032aa3dcc14f55caa38a9b88652b2f9f
(Done for all reentrancy comments)